### DISCUSSION TITLE
Getting debug feedback on mapping errors?

### DISCUSSION THREAD

**Discussion by clansdown** [MorphiaOrg/morphia#1861] [Q&A]
I'm using morphia 1.6.1 and getting an error when saving an object where an array field isn't null, but the error is not helpful ("dev.morphia.mapping.MappingException: Error mapping field:[field name]")

But the field referred to there is an embedded object and normally maps fine, unless one of its array fields is filled out. That field is itself an array of objects, etc.

Is there any way to get morphia to tell me what the actual problem is, i.e. which field of which object is actually causing the problem, and not merely that the problem was encountered in the one significant field of a db container object?

The relevant part of the traceback is:
	Mapper.java:1039 (toDBObject)
	Mapper.java:636 (toDBObject)
	DatastoreImpl.java:1421 (entityToDBObj)
	DatastoreImpl.java:1289 (save)
	DatastoreImpl.java:773 (save)
	DatastoreImpl.java:756 (save)

**Comment by evanchooly** [comment]
Is there no "caused by" section in the stack trace?  I'll pull up that code and see if it's suggestive of anything but i'll likely need more to go on.  If you could put together a [reproducer](/morphiaorg/reproducer) I'd be happy to take a look and see.
